src/components/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../PeopleFilters';
+import { Loader } from '../Loader';
+import { PeopleTable } from '../PeopleTable';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { filteredPeople } from '../../utils/filteredPeople';
+
+export const PeoplePage = () => {
+  const [peopleFromServer, setPeopleFromServer]
+    = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [searchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+  const centuries = searchParams.getAll('centuries') || '';
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const isResponseSuccessful = peopleFromServer && !isLoading && !errorMessage;
+
+  const preparedPeopleData = (people: Person[]) => {
+    const preparedPeople = people.map((person) => {
+      let mother;
+      let father;
+
+      if (person.motherName) {
+        mother = people.find(({ name }) => person.motherName === name);
+      }
+
+      if (person.fatherName) {
+        father = people.find(({ name }) => person.fatherName === name);
+      }
+
+      return { ...person, mother, father };
+    });
+
+    setPeopleFromServer(preparedPeople);
+  };
+
+  const filterPeople = useMemo(() => {
+    return filteredPeople(peopleFromServer, {
```

Comment:
just FYI: should be vice versa - const `filteredPeople` with array and function `filterPeople`. Function does some action so when I see verb in name I expect it's a function